WordPress/src/main/java/org/wordpress/android/localcontentmigration/LocalMigrationError.kt

```diff
@@ -6,7 +6,7 @@ import org.wordpress.android.localcontentmigration.LocalContentEntityData.Compan
 sealed class LocalMigrationError {
     sealed class ProviderError : LocalMigrationError() {
         data class NullValueFromQuery(val forEntity: LocalContentEntity) : ProviderError()
-        data class NullCursor(val forEntity: LocalContentEntity) : ProviderError()
+        data class NullCursor(val forEntity: LocalContentEntity, val throwable: Throwable? = null) : ProviderError()
```

Comment:
Maybe we should create a new `ProviderError` class for this (rather than repurposing `NullCursor` 🤔 ? Wdyt?

If you think this makes sense, I've added a branch with this suggestion [here](https://github.com/wordpress-mobile/WordPress-Android/compare/fix/18627-SecurityException...fix/18627-SecurityException--suggestion-new-error-class).

Reply:
> Maybe we should create a new ProviderError class for this (rather than repurposing NullCursor 🤔 

Thank you for your suggestion @mkevins! This makes sense 👍 
I've merged you branch on this PR.

Reply:
Awesome!